hazelcast/src/main/java/com/hazelcast/map/impl/wan/WanMapEntryView.java

```diff
@@ -0,0 +1,252 @@
+/*
+ * Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.map.impl.wan;
+
+import com.hazelcast.core.EntryView;
+import com.hazelcast.map.impl.MapDataSerializerHook;
+import com.hazelcast.nio.IOUtil;
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.nio.serialization.BinaryInterface;
+import com.hazelcast.nio.serialization.IdentifiedDataSerializable;
+
+import java.io.IOException;
+
+/**
+ * WAN heap based implementation of {@link EntryView} for keeping
+ * compatibility when sending to older (3.8+) clusters.
```

Comment:
Maybe it is worth adding a comment that this implementation intentionally breaks the `DataSerializable` conventions - what the broken tests are worrying about - for the sake of solving the mentioned WAN compatibility issue.